app/models/prosecution_case.rb

```diff
@@ -1,5 +1,7 @@
 # frozen_string_literal: true
 
+# WARNING! Despite the name, a ProsecutionCase DB record can contain _either_
+# a prosecution case our a court application record
```

Comment:
👍 for the comment!

that's a good point, I was wondering if we can clarify the distintion in the code,
maybe adding an abstraction, called `LegalCase` or something,
and 
`ProsecutionCase < LegalCase`

`CourtApplication < LegalCase`

and we can associate LegalCase with the existing `prosecution_cases` table

could that help? 🤔

Reply:
Great idea! I have done exactly that.